pkg/querier/codec/instrumented_codec.go

```diff
@@ -0,0 +1,57 @@
+package codec
+
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	v1 "github.com/prometheus/prometheus/web/api/v1"
+	"github.com/weaveworks/common/middleware"
+)
+
+type InstrumentedCodecMetrics struct {
+	responseSizeHistogram *prometheus.HistogramVec
+}
+
+func NewInstrumentedCodecMetrics(reg prometheus.Registerer) *InstrumentedCodecMetrics {
+	return &InstrumentedCodecMetrics{
+		responseSizeHistogram: promauto.With(reg).NewHistogramVec(prometheus.HistogramOpts{
+			Namespace:                       "cortex",
+			Name:                            "querier_codec_response_size",
```

Comment:
Is this the size before compression or after compression?

Is it better to remove `codec` in the name? This sounds quite confusing.

Reply:
this is before compression (hence i added the codec in the name i guess...)